lib/charms/mysql/v0/s3_helpers.py

```diff
@@ -86,7 +86,8 @@ def upload_content_to_s3(content: str, content_path: str, s3_parameters: Dict) -
         )
         return False
     finally:
-        ca_file.close()
+        if ca_file:
+            ca_file.close()
```

Comment:
Why don't we close right after ca_file.flush() ? Do you also want to close `temp_file`?
Anyway, this will work.

Reply:
The `NamedTemporaryFile` will be deleted once closed, hence closing it by the end.
The `temp_file` is used within a context manager (`with`), which closes it when exits the context